app/controllers/ReadProtectionsController.scala

```diff
@@ -85,6 +85,18 @@ class ReadProtectionsController @Inject() (
       plaConnector.readProtections(nino).map(_.map(TransformedReadResponseModel.from))
     }
 
+  def fetchFixedProtection2016(nino: String)(implicit hc: HeaderCarrier): Future[Option[ProtectionModel]] =
+
+    fetchProtections(nino).flatMap {
+
+      case Right(transformedReadResponseModel: TransformedReadResponseModel) =>
+        Future.successful(filterOnlyFixedProtection2016(transformedReadResponseModel))
```

Comment:
`flatMap` is redundant here - you can use `map` and just don't wrap results in `Future`.